packages/docs/guide/advanced/data-fetching.md

```diff
@@ -27,9 +29,54 @@ Let's assume we have a `Post` component that needs to fetch the data for a post
     </div>
   </div>
 </template>
+
+<script setup>
+import { ref, watch } from 'vue'
+import { useRoute } from 'vue-router'
+import { getPost } from './api.js'
+
+const route = useRoute()
+
+const loading = ref(false)
+const post = ref(null)
+const error = ref(null)
+
+// watch the params of the route to fetch the data again
+watch(() => route.params.id, fetchData, { immediate: true })
+
+async function fetchData() {
```

Comment:
let's add the id as a paremeter rather than relying on the route instance within the function